bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerHttpApiV100.java

```diff
@@ -246,49 +286,118 @@ protected String getRequestRequiredOptions() {
         String returnContent;
-
+        JcResponse resp;
         try {
             returnContent = http.sendHttpGet(getBaseUrl() + CMD_STATUS_INFO, getRequestRequiredOptions());
-        } catch (CommunicationApiException exp) {
+            resp = gson.fromJson(returnContent, JcResponse.class);
+            if (resp == null) {
+                throw new CommunicationApiException(
+                        "There was a problem in the HTTP communication: jcReply was empty.");
+            }
+        } catch (CommunicationApiException | JsonSyntaxException exp) {
             throw new CommunicationApiException(
                     "There was a problem in the HTTP communication with the OpenSprinkler API: " + exp.getMessage());
         }
-
-        JcResponse resp = gson.fromJson(returnContent, JcResponse.class);
         return resp;
     }
 
-    private static class JcResponse {
-        public List<List<Integer>> ps;
-        @SerializedName(value = "sn1", alternate = "rs")
-        public int rs;
-        public Integer curr;
+    private JoResponse getOptions() throws CommunicationApiException, UnauthorizedApiException {
+        String returnContent;
+        JoResponse resp;
+        try {
+            returnContent = http.sendHttpGet(getBaseUrl() + CMD_OPTIONS_INFO, getRequestRequiredOptions());
+            resp = gson.fromJson(returnContent, JoResponse.class);
+            if (resp == null) {
+                throw new CommunicationApiException(
+                        "There was a problem in the HTTP communication: joReply was empty.");
+            }
+        } catch (CommunicationApiException | JsonSyntaxException exp) {
+            throw new CommunicationApiException(
+                    "There was a problem in the HTTP communication with the OpenSprinkler API: " + exp.getMessage());
+        }
+        return resp;
     }
 
-    private JoResponse getOptions() throws CommunicationApiException {
+    protected JsResponse getStationStatus() throws CommunicationApiException, UnauthorizedApiException {
         String returnContent;
-
+        JsResponse resp;
         try {
-            returnContent = http.sendHttpGet(getBaseUrl() + CMD_OPTIONS_INFO, getRequestRequiredOptions());
-        } catch (CommunicationApiException exp) {
+            returnContent = http.sendHttpGet(getBaseUrl() + CMD_STATION_INFO, getRequestRequiredOptions());
+            resp = gson.fromJson(returnContent, JsResponse.class);
+            if (resp == null) {
+                throw new CommunicationApiException(
+                        "There was a problem in the HTTP communication: jsReply was empty.");
+            }
+        } catch (CommunicationApiException | JsonSyntaxException exp) {
             throw new CommunicationApiException(
                     "There was a problem in the HTTP communication with the OpenSprinkler API: " + exp.getMessage());
```

Comment:
Same issue here. It looks like there are bunch of places with that issue so please just fix it wherever it may apply.

Reply:
DONE